## Homework 3
### Friday September 11th 2020

**Question #1**

In [1]:
terabyte= 2**40

In [2]:
song= 8*2**20

In [3]:
terabyte//song

131072

the hard drive can hold 131,072 songs.

**Question #2**

In [4]:
#a)
15*3-2

43

The value of a is 43 because we multiply 15 by 3 first to get 45 then subtract 2 from the result.

In [5]:
#b)
15-3*2

9

The value of this is 9 because we multiply 3 by 2 first to get 6, then subtract six from 15 to get 9

In [6]:
#c)
15*3//2

22

The answer is 22 because we multiply 15 and 3 first to get 45, then 2 goes into 45 22 times with a remainedr of 1. So teh answer is 22.

In [7]:
#d)
15*3/2

22.5

The answer is 22.5 because wemultiply first then divide. 45 divided by 2 is 22.5

In [8]:
#e)
15*3%2

1

In #c we found that 2 goes into 45(which is 15*3) 22 times with a remainder of 1. Mod gives the remainder after division which is 1 in this case

In [9]:
#f)
15*3/2e0

22.5

This expression's answer is 22.5 because we multiply 15 by 3 first then divide by 2. This case is different from #d because 2 is written as a float. If the decimals were more than 16, python would have rounded them to 16 decimal points.

**Question #3**

As shown below, the code x=y and y=x assigns the same value to both x and y. This is because the first code x=y makes both variables have the same value. 

If x=cow and y=goat, we can swap the values by:

1) equate x,y to y,x. This will prevent the problem of assigning one first then the other which makes them both equal to one value. As shown below, this method works unlike the first one.

In [10]:
x="cow"
y="goat"
x=y
y=x

print(x)
print(y)

goat
goat


In [11]:
x="cow"
y="goat"

In [12]:
x,y=y,x

In [13]:
print(y)

cow


In [14]:
print(x)

goat


**Question #4**

In [3]:
Myemployees= ["Muthoni", "Makena", "Wangui", "Waithegeni", "Nyawira", "Nyaguthii", "Kago", "Warui","Gitonga","Ndururi"]

Daysopen= ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]


for employee in Myemployees:   
    dayindex=Myemployees.index(employee)%5
    print(employee, "works on",Daysopen[dayindex])
    

Muthoni works on Monday
Makena works on Tuesday
Wangui works on Wednesday
Waithegeni works on Thursday
Nyawira works on Friday
Nyaguthii works on Monday
Kago works on Tuesday
Warui works on Wednesday
Gitonga works on Thursday
Ndururi works on Friday


In [4]:

for day in Daysopen:
    print("on",day,":",Myemployees[Daysopen.index(day)],"and",Myemployees[Daysopen.index(day)+5])

on Monday : Muthoni and Nyaguthii
on Tuesday : Makena and Kago
on Wednesday : Wangui and Warui
on Thursday : Waithegeni and Gitonga
on Friday : Nyawira and Ndururi


In [5]:

##Here is my program. Replace the inputs with anything and run the code with your personal inputs.

myemployeeslist=input(str)
myemployeeslist=myemployeeslist.split()

<class 'str'> ali ben bob jake


In [6]:
daysoftheworkweek=input(str)
daysoftheworkweek=daysoftheworkweek.split()

<class 'str'> mon tues


In [7]:
for day in daysoftheworkweek:
    print("on",day, myemployeeslist[daysoftheworkweek.index(day)],"and", myemployeeslist[daysoftheworkweek.index(day)+5]) 

IndexError: list index out of range

In [8]:
DaysOpen = input("Enter the days you are open:")

Employees= input("Enter the Employees who are Working for You:")

for employee in range(10):
    DaysIndex = employee%5
    print(Employees.split(",")[employee], "is working on ", DaysOpen.split(",")[DaysIndex]   )



Enter the days you are open: mon tues
Enter the Employees who are Working for You: aaron bob dave kelly


aaron bob dave kelly is working on  mon tues


IndexError: list index out of range